src/js/line.js

```diff
@@ -0,0 +1,521 @@
+            return new Date(d);
+        });
+
+        return d3.time.scale()
+            .domain([minDate, maxDate])
+            .range([padding, width - padding * 2]);
+    };
+
+    floe.chartAuthoring.lineChart.chart.getYAxis = function (that) {
+        var width = that.options.svgOptions.width;
+        var padding = that.options.lineOptions.padding;
+        var yScale = floe.chartAuthoring.lineChart.chart.getYScale(that);
+
+        var yAxis = d3.svg.axis()
+            .orient("left")
+            .scale(yScale)
+            .innerTickSize(- width + padding * 3)
+            .outerTickSize(0)
+            .tickPadding(10);
+
+        return yAxis;
+    };
+
+    floe.chartAuthoring.lineChart.chart.getXAxis = function (that) {
+        var xScale = floe.chartAuthoring.lineChart.chart.getXScale(that),
+            numberOfXAxisTicks = that.options.lineOptions.numberOfXAxisTicks;
+
+        // See https://github.com/mbostock/d3/wiki/Time-Formatting for
+        // explanation of how time formatting works in D3
+        var customTickFormat = d3.time.format.multi([
+            [".%L", function (d) { return d.getMilliseconds(); }],
+            [":%S", function (d) { return d.getSeconds(); }],
+            ["%I:%M", function (d) { return d.getMinutes(); }],
+            ["%I %p", function (d) { return d.getHours(); }],
+            ["%a %d", function (d) { return d.getDay() && d.getDate() !== 1; }],
+            ["%b %d", function (d) { return d.getDate() !== 1; }],
+            ["%b", function (d) { return d.getMonth(); }],
+            ["%Y", function () { return true; }]
+        ]);
+
+        var xAxis = d3.svg.axis()
+            .tickFormat(customTickFormat)
+            .ticks(numberOfXAxisTicks)
+            .orient("bottom")
+            .scale(xScale);
+
+        return xAxis;
+
+    };
+
+    floe.chartAuthoring.lineChart.chart.getLineGenerator = function (that) {
+        var interpolation = that.options.lineOptions.interpolation,
+            yScale = floe.chartAuthoring.lineChart.chart.getYScale(that),
+            xScale = floe.chartAuthoring.lineChart.chart.getXScale(that);
+
+        var line = d3.svg.line()
+            .interpolate(interpolation)
+            .x(function (d) {
+                return xScale(new Date(d.date));
+            })
+            .y(function (d) {
+                return yScale(d.value);
+            });
+
+        return line;
+    };
+
+    floe.chartAuthoring.lineChart.chart.getAreaGenerator = function (that) {
+        var interpolation = that.options.lineOptions.interpolation;
+        var height = that.options.svgOptions.height,
+            padding = that.options.lineOptions.padding,
+            yScale = floe.chartAuthoring.lineChart.chart.getYScale(that),
+            xScale = floe.chartAuthoring.lineChart.chart.getXScale(that);
+
+        var area = d3.svg.area()
+            .interpolate(interpolation)
+            .x(function (d) {
+                return xScale(new Date(d.date));
```

Comment:
Refactor so that x and y are not hardcoded to accept members named "date" and "value", and that the former is not required to be always converted to a Date